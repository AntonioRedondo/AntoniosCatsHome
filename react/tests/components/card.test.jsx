import React from "react";
import Renderer from "react-test-renderer";
import "jest-styled-components";

import Card from "../../src/js/components/card.jsx";
import v from "../../src/js/style/variables";


describe("Given the Card component", () => {
	let component;
	let tree;
	
	describe("When the component is mounted", () => {
		beforeEach(() => {
			component = Renderer.create(
				<Card
					id="2"
					name="Fluffy"
					onClick={ function() {} } />
			);
			tree = component.toJSON();
		});
		
		it("Should render children", () => {
			expect(tree).toMatchSnapshot();
		});
	});
	
	describe("When the card is clicked", () => {
		it("Should select the card", () => {
			expect(tree).toHaveStyleRule("background", v.grayE);
			
			component = Renderer.create(
				<Card
					id="2"
					name="Fluffy"
					onClick={ function() {} }
					itemSelected="2" />
			);
			tree = component.toJSON();
			
			expect(tree).toHaveStyleRule("background", "papayawhip");
		});
	});
	
});
