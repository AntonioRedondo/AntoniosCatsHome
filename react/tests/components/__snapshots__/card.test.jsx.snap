// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Card component When the component is mounted Should render children 1`] = `
.c0 {
  width: 100%;
  height: 250px;
  margin: 0 0 40px;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #eeeeee;
  box-shadow: 0 -5px 0 #b74b66,0 0 10px rgba(0,0,0,0.5);
  color: #333333;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
}

.c0:hover {
  background: white;
  box-shadow: 0 -5px 0 orange,10px 10px 30px rgba(0,0,0,0.2);
  cursor: pointer;
  -webkit-transform: scale(1.025);
  -ms-transform: scale(1.025);
  transform: scale(1.025);
}

.c0:active {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  background: lightyellow;
}

.c1 {
  width: 250px;
  height: 250px;
  background-image: url('img/2.jpg');
  background-size: cover;
  position: relative;
}

.c2 {
  padding: 40px;
  overflow: hidden;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: 1em;
}

.c3 {
  margin: 0;
  font-family: "Rokkitt",serif;
  font-size: 2.5em;
}

.c4 {
  margin: 0 0 20px;
  color: #666666;
}

@media (max-width:800px) {
  .c0 {
    height: 150px;
    margin: 0;
    box-shadow: 0 5px 0 #b74b66 inset,10px 10px 30px rgba(0,0,0,0.2);
  }
}

@media (max-width:800px) {
  .c1 {
    width: 150px;
    min-width: 150px;
    height: 150px;
  }
}

@media (max-width:800px) {
  .c2 {
    padding-right: 0;
  }
}

@media (max-width:800px) {
  .c4 {
    display: none;
  }
}

<article
  className="c0"
  onClick={undefined}
>
  <div
    className="c1"
    id="2"
  />
  <div
    className="c2"
  >
    <h1
      className="c3"
    >
      Fluffy
    </h1>
    <div
      className="c4"
    />
  </div>
</article>
`;
